src/func/Microsoft.Health.Fhir.Ingest.Host/Startup.cs

```diff
@@ -22,16 +22,41 @@ public override void Configure(IFunctionsHostBuilder builder)
         {
             EnsureArg.IsNotNull(builder, nameof(builder));
 
-            builder.Services.AddSingleton<ITelemetryLogger>(sp =>
+            var instrumentationKey = Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY");
+
+            if (instrumentationKey != null)
+            {
+                var configDescriptor = builder.Services.SingleOrDefault(tc => tc.ServiceType == typeof(TelemetryConfiguration));
+                if (configDescriptor?.ImplementationFactory == null)
+                {
+                    return;
+                }
+
+                var implFactory = configDescriptor.ImplementationFactory;
+
+                builder.Services.Remove(configDescriptor);
+                builder.Services.AddSingleton<ITelemetryLogger>(provider =>
+                {
+                    if (!(implFactory.Invoke(provider) is TelemetryConfiguration config))
+                    {
+                        return null;
```

Comment:
What happens in the result of the singleton factory is null? It seems like we will just have no logger then.  It seems like we should either throw an exception or proceed with the default configuration like you do when the instrumentation key is null.

Reply:
Updated to throw exception